src/main/cli/cli.c

```diff
@@ -1362,6 +1362,66 @@ static void cliSerialPassthrough(char *cmdline)
 }
 #endif
 
+#if defined(USE_ESC_SENSOR)
+static void cliEscSensorPassthrough(char *cmdline)
+{
+    uint32_t useEscSensorBaud = 115200;
+    if (!isEmpty(cmdline)) {
+        useEscSensorBaud = atoi(cmdline);
+    }
+    
+    const serialPortConfig_t *portConfig = findSerialPortConfig(FUNCTION_ESC_SENSOR);
+    if (!portConfig) {
+        cliPrintLine("no esc sensor port configured.");
+        return;
+    }
+    
+    serialPortUsage_t *serialPortUsage = findSerialPortUsageByIdentifier(portConfig->identifier);
+    serialPortUsage->function = FUNCTION_NONE;
+    serialPortUsage->serialPort = NULL;
+    
+    serialPort_t *passThroughPort;
+    passThroughPort = openSerialPort(portConfig->identifier,
+        FUNCTION_NONE,
+        NULL,
+        NULL,
+        useEscSensorBaud,
+        MODE_RXTX,
+        (SERIAL_STOPBITS_1 | SERIAL_PARITY_NO | SERIAL_BIDIR | SERIAL_BIDIR_PP)
+    );
+    
+    if (!passThroughPort) {
+        cliPrintLine("esc sensor port could not be opened.");
+        return;
+    }
+    
+    pwmDisableMotors();
+    pwmOutputPort_t *pwmMotors = pwmGetMotors();
+    for (volatile uint8_t i = 0; i < MAX_SUPPORTED_MOTORS; i++) {
+        if (pwmMotors[i].enabled && pwmMotors[i].io != IO_NONE) {
+            const ioTag_t tag = motorConfig()->dev.ioTags[i];
```

Comment:
Not sure why you are doubling up here. `pwmMotors[i].io` is already set from `motorConfig()->dev.ioTags[i]` in `motorDevInit()`, so going back to it seems to be moot.

Reply:
i thought its to check if there is a timer hardware (which i dont need)

```
const timerHardware_t *timerHardware = timerGetByTag(tag);
                if (timerHardware) {
```
i took the whole thing from escserial stuff. so i will try without.

Reply:
This should still be addressed.

Reply:
i think i got now what you mean :) is it correct now?

Reply:
Yeah, much better now.